srv/modules/runners/rescinded.py

```diff
@@ -68,6 +73,73 @@ def osds(cluster='ceph'):
             log.debug(data[minion])
     return list(set(_ids))
 
+
+def orphaned_hosts(cluster='ceph'):
+    """
+    List any orphaned host entries in the CRUSH map
+    """
+    search = "I@cluster:{} and I@roles:master".format(cluster)
+    local = salt.client.LocalClient()
+    data = local.cmd(search, 'osd.tree_from_master', [], tgt_type="compound")
+    osd_tree = data[list(data.keys())[0]]
+    subtree = [x for x in osd_tree['nodes']
+               if x['type'] == 'host' and len(x['children']) == 0]
+    orphaned_hosts_list = [x['name'] for x in subtree]
+    if orphaned_hosts_list:
+        log.debug("rescinded.orphaned_hosts: found orphaned CRUSH hosts ->{}<-"
+                  .format(','.join(orphaned_hosts_list)))
+    else:
+        log.debug("osd.orphaned: no orphaned CRUSH hosts found")
+    assert isinstance(orphaned_hosts_list, list),\
+        "List of orphaned CRUSH host entries is not a list"
+    return orphaned_hosts_list
+
+
+def nodes(cluster='ceph', **kwargs):
+    """
+    List the short hostnames of minions that are not storage nodes
+    and whose corresponding CRUSH host buckets are now orphaned
+    (there should be only one, but handle multiple)
```

Comment:
There can be multiple if a user removes multiple osd nodes from the policy.cfg and executes stage.5